twenty-first/src/shared_math/b_field_element.rs

```diff
@@ -528,6 +530,35 @@ impl PrimitiveRootOfUnity for BFieldElement {
     }
 }
 
+impl BFieldElement {
+    pub fn emojify(&self) -> String {
+        let data: [u8; 8] = (*self).into();
+        let emojis = emojihash::hash(&data);
+
+        let mut rv = String::new();
+        let mut max_len = BFE_EMOJI_LEN;
+        for e in emojis.chars() {
+            if 0 == max_len {
+                break;
+            }
+            rv.push(e);
+            max_len -= 1;
+        }
+        rv
+    }
+}
+
+impl Emojible for BFieldElement {
+    fn to_emoji(&self) -> String {
+        let prefix = "[";
+        let suffix = "]";
+        let mut rv = String::from(prefix);
+        rv.push_str(&self.emojify());
+        rv.push_str(suffix);
+        rv
+    }
+}
```

Comment:
One is for `raw` emojis, but we want different ornamentation on them depending on the are part of a XFE or BFE - so we need three functions. I've added comments to clear this up.

Reply:
This is over-complicated.

You can print the emojihash of a `BFieldElement` in one way, and you can print the emojihash of an `XFieldElement` in one way, and one printer can depend on the other without differentiating between "raw" and "ornamental". This reflects how the `Display` instances work: `BFieldElement`'s is derived for and `XFieldElement`'s depends on it:

```rust
impl Display for XFieldElement {
    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
        if self.coefficients[2].is_zero() && self.coefficients[1].is_zero() {
            write!(f, "{}_xfe", self.coefficients[0])
        } else {
            write!(
                f,
                "({:>020}·x² + {:>020}·x + {:>020})",
                self.coefficients[2], self.coefficients[1], self.coefficients[0],
            )
        }
    }
}
```

Reply:
This works because you rely on the `Display` instance of `BFieldElement` when you print `self.coefficients[0]` in the `else` clause, and this does not add any ornamentation. Had it printed "[{}]" instead of just "{}", you would have had the same problem.